hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/JmsIntegrationTestBase.java

```diff
@@ -383,33 +384,50 @@ public void sinkTopic_whenBuilder_withParameters() throws JMSException {
 
     @Test
     public void stressTest_exactlyOnce_forceful() throws Exception {
-        stressTest(false, EXACTLY_ONCE);
+        stressTest(false, EXACTLY_ONCE, false);
     }
 
     @Test
     public void stressTest_exactlyOnce_graceful() throws Exception {
-        stressTest(true, EXACTLY_ONCE);
+        stressTest(true, EXACTLY_ONCE, false);
     }
 
     @Test
     public void stressTest_atLeastOnce_forceful() throws Exception {
-        stressTest(false, AT_LEAST_ONCE);
+        stressTest(false, AT_LEAST_ONCE, false);
     }
 
     @Test
     public void stressTest_noGuarantee_forceful() throws Exception {
-        stressTest(false, NONE);
+        stressTest(false, NONE, false);
     }
 
-    private void stressTest(boolean graceful, ProcessingGuarantee maxGuarantee)
+    @Test
+    public void stressTest_exactlyOnce_forceful_durableTopic() throws Exception {
+        stressTest(false, EXACTLY_ONCE, true);
+    }
+
+    private void stressTest(boolean graceful, ProcessingGuarantee maxGuarantee, boolean useTopic)
```

Comment:
I frequently see that we always keep adding parameters to test in this way and it makes it harder to read the test with a lot of conditional logic inside. It would be better if they were refactored in a way that would avoid conditional test logic.